src/Entities/Frames.jl

```diff
@@ -38,9 +38,12 @@ struct Frame{dim, T <: Real, N <: AbstractNode{dim, T},
         new{dim, T, N, VN, G}(nodes, cross_section, mass_matrix, Symbol(label))
     end
 end
-function Frame(n₁::N, n₂::N, g::G, mass_matrix::MassMatrix = Consistent,
-        label::Label = NO_LABEL) where {N <: AbstractNode, G <: AbstractCrossSection}
-    Frame(SVector(n₁, n₂), g, mass_matrix, label)
+function Frame(n₁::N1, n₂::N2, g::G, mass_matrix::MassMatrix = Consistent,
+        label::Label = NO_LABEL) where {N1 <: AbstractNode, N2 <: AbstractNode, G <: AbstractCrossSection}
+        temp = promote(n₁.x,n₂.x)
+        n1r = Node(temp[1],n₁.dofs)
```

Comment:
Note that we will always allocate a new node, so maybe we can so it only when the numeric types missmatch?

Reply:
If a new node is allocated as proposed, does the performance reduce? I don't know how to check wether the numeric types are mismatch. Please help me.

Reply:
Sure, the cleanest way is to extract the types directly from the `Node` struct as is parametric in the `Node` type:

``` 
julia> n1
• Node at [1.2, 2.3, 3.4] and dofs {}
julia> typeof(n1)
Node{3, Float64}
julia> n2
• Node at [2, 3, 4] and dofs {}
julia> typeof(n2)
Node{3, Int64}
```

So this  should work
```
function Frame(n1::N1, n2::N2, g::G, mass_matrix::MassMatrix = Consistent,
        label::Label = NO_LABEL) where {dim, T1, T2, N1 <: AbstractNode{dim, T1},
        N2 <: AbstractNode{dim, T2}, G <: AbstractCrossSection}
    if T1 == T2
        Frame(SVector(n1, n2), g, mass_matrix, label)
    else
        temp = promote(n1.x, n2.x)
        n1r = Node(first(temp), n1.dofs)
        n2r = Node(last(temp), n2.dofs)
        Frame(SVector(n1r, n2r), g, mass_matrix, label)
    end
end

``

Note: It's hard to me to reproduce this branch as is not created with git so I can't fetch your changes locally.so I recommend next time https://github.com/apps/desktop so I don't have to copy paste all your changes locally if possible :)   I'm happy to pair on if u find any trouble while you do the set up.

Reply:
Sure the cleanest way is to extract the types form `Node` object as it is parametric in the coords type:

```
julia> n2
• Node at [2, 3, 4] and dofs {}

julia> typeof(n2)
Node{3, Int64}
``` 
So we can use:
```
function Frame(n1::N1, n2::N2, g::G, mass_matrix::MassMatrix = Consistent,
        label::Label = NO_LABEL) where {dim, T1, T2, N1 <: AbstractNode{dim, T1},
        N2 <: AbstractNode{dim, T2}, G <: AbstractCrossSection}
    if T1 == T2
        Frame(SVector(n1, n2), g, mass_matrix, label)
    else
        temp = promote(n1.x, n2.x)
        n1r = Node(first(temp), n1.dofs)
        n2r = Node(last(temp), n2.dofs)
        Frame(SVector(n1r, n2r), g, mass_matrix, label)
    end
end
```

Next time it would be easier for me if you create this branch from your local desktop git https://github.com/apps/desktop if not I have to copy and paste your chages to reproduce what you did. I'm happy to help if u found any trouble with this.